File: RRComSSys/Projects/CVM/RRComSSys.CVM/src/edu/fiu/cis/rrcomssys/cvm/mte/xcmlmodel/Device.java
package edu.fiu.cis.rrcomssys.cvm.mte.xcmlmodel;

import java.util.LinkedList;
import java.util.List;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;

public class Device
{
	/* Attributes */
	@Attribute(name = "deviceID", required = true)
	private String _id = "";

	@Attribute(name = "isLocal", required = true)
	private boolean _isLocal = false;

	@Attribute(name = "isVirtual", required = true)
	private boolean _isVirtual = false;

	/* Elements */
	@ElementList(entry = "deviceCapability", inline = true, required = true)
	private List<CapabilityType> _capabilitiesList = new LinkedList<CapabilityType>();

	/* Constructors & Initializers */
	public Device()
	{}

	/* Properties */
	public String getID()
	{
		return _id;
	}

	public void setID(String value)
	{
		_id = value;
	}

	public boolean isLocal()
	{
		return _isLocal;
	}

	public void setIsLocal(boolean value)
	{
		_isLocal = value;
	}

	public boolean isVirtual()
	{
		return _isVirtual;
	}

	public void setIsVirtual(boolean value)
	{
		_isVirtual = value;
	}

	public List<CapabilityType> getCapabilities()
	{
		return _capabilitiesList;
	}
}

File: trunk/RRComSSys/Projects/CVM/RRComSSys.CVM.ObjectModel/XCMLModel/Device.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace RRComSSys.CVM.ObjectModel.XCMLModel
{
	public class Device : IXCMLItem
	{
		#region Member Variables
		private String _id = "";
		private bool _isLocal = false;
		private bool _isVirtual = false;
		private List<CapabilityType> _capabilitiesList = new List<CapabilityType>();
		#endregion

		#region Properties
		[XmlAttribute(AttributeName = "deviceID")]
		public String ID
		{
			get { return _id; }
			set { _id = value; }
		}

		[XmlAttribute(AttributeName = "isLocal")]
		public bool IsLocal
		{
			get { return _isLocal; }
			set { _isLocal = value; }
		}

		[XmlAttribute(AttributeName = "isVirtual")]
		public bool IsVirtual
		{
			get { return _isVirtual; }
			set { _isVirtual = value; }
		}

		[XmlElement(ElementName = "deviceCapability")]
		public List<CapabilityType> Capabilities
		{
			get { return _capabilitiesList; }
			set { _capabilitiesList = value; }
		}
		#endregion
	}
}
